src/Component/Metadata/Resource/Factory/AttributesResourceMetadataCollectionFactory.php

```diff
@@ -136,6 +136,17 @@ private function getOperationWithDefaults(ResourceMetadata $resource, Operation
             $operation = $operation->withRepository($resourceConfiguration->getServiceId('repository'));
         }
 
+        if (null === $operation->getFormType()) {
+            $formType = $resource->getFormType() ?? $resourceConfiguration->getClass('form');
+            $operation = $operation->withFormType($formType);
+        }
+
+        if (null === $operation->getFormOptions()) {
+            $operation = $operation->withFormOptions([
+                'data_class' => $resourceConfiguration->getClass('model'),
+            ]);
+        }
```

Comment:
This again seems to be a good fit to extract to smaller objects, which can specialize in mutation of operations itself, as this class started to grow significantly